backends/printing/printman.h

```diff
@@ -0,0 +1,34 @@
+/* ScummVM - Graphic Adventure Engine
+ *
+ * ScummVM is the legal property of its developers, whose names
+ * are too numerous to list here. Please refer to the COPYRIGHT
+ * file distributed with this source distribution.
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ *
+ */
+
+
+#include "common/scummsys.h"
+#include "common/str.h"
+
+class PrintingManager {
+public:
+	virtual ~PrintingManager() = 0;
+
+	virtual void printImage(Common::String jobName, byte *pixels, byte *palette, uint32 width, uint32 height) = 0;
+	void printImage(byte * pixels, byte * palette, uint32 width, uint32 height) {
```

Comment:
This API is extremely limited. Basically, here you're printing only 8 bit images which is way too limited.

My suggestion is to pass here a `Graphics::ManagedSurface()` which could be of any format and even a palette attached to it.

Reply:
I thought I looked at that. But I will take a second look. I'm not experienced with the graphics subsystem, so I must've overlooked it.

Reply:
`Graphics::ManagedSurface()` has `setPalette()` method.

In this particular case, with the WIZ images, you need to create a CLUT8 surface, copy over the pixels and set the palette. Should be like 2 loops and a single call

Reply:
Yeah, I'm seeing that now.

Btw, you happen to know the comp == 1 format for wiz images? I can't make heads or tails out of it. The Let's Explore series is printing in that format.

Reply:
Did you see line 1621 in `wiz_he.cpp`?  https://github.com/scummvm/scummvm/blob/master/engines/scumm/he/wiz_he.cpp#L1621

Reply:
I did see that one yes. Sadly didn't make things clearer. I will take a third look at it later.